### PR TITLE
Add transport_worker_dispatcher

### PR DESCRIPTION
**Pull request by jakobod**
This PR adds a transport_worker_dispatcher which dispatches calls to transport_workers.

### CODE REVIEW

**Comment by Neverlord** [comment]
> Refactor resolve() to use caf::uri

What motivated this change? From my understanding so far, we would assign an URI (e.g. `tcp://mars:8080`)  to the endpoint manager responsible for that connection. Then, resolve only needs the path to build the full URI when creating proxies (e.g. `tcp://mars:8080/foo/bar`).

**Comment by jakobod** [comment]
I was talking about this with @josephnoir because I had problems getting the `node_id` which I needed in `resolve`. I probably misunderstood him when he told me to use an `uri` in the `resolve` call.
The `uri` is still missing in the `endpoint_manager`. I can just revert that commit and add the `uri` to implement the `resolve` call properly. That seems to make more sense :+1:.

File: libcaf_net/caf/net/transport_worker_dispatcher.hpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  transport_worker_dispatcher(factory_type factory)
+    : factory_(std::move(factory)) {
+    // nop
+  }
+
+  // -- member functions -------------------------------------------------------
+
+  template <class Parent>
+  error init(Parent& parent) {
+    for (const auto& p : workers_by_id_) {
```
[the remark this answers is not in the document]
Comment (jakobod) [reply libcaf_net/caf/net/transport_worker_dispatcher.hpp]: You're right. I would add a parameter `Parent` to `add_new_worker` and call `init` there.

File: libcaf_net/caf/net/transport_worker_dispatcher.hpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      auto worker = p.second;
+      worker->handle_error(error);
+    }
+  }
+
+  template <class Parent>
+  void add_new_worker(Parent& parent, node_id node, id_type id) {
+    auto application = factory_.make();
+    auto worker = std::make_shared<worker_type>(std::move(application), id);
+    worker->init(parent);
```
Comment (Neverlord) [inline libcaf_net/caf/net/transport_worker_dispatcher.hpp]: This can fail:

```suggestion
    if (auto err = worker->init(parent)) return err;
```

**Pull request merged**